library/core/functions.general.php

```diff
@@ -3859,6 +3879,37 @@ function slugify($text) {
     }
 }
 
+if (!function_exists('sprintft')) {
+    /**
+     * A version of {@link sprintf()} That translates the string format.
+     *
+     * @param string $formatCode The format translation code.
+     * @param mixed $arg1 The arguments to pass to {@link sprintf()}.
+     * @return string The translated string.
+     */
+    function sprintft($formatCode, $arg1 = null) {
+        $args = func_get_args();
+        $args[0] = t($formatCode, $formatCode);
+        return call_user_func_array('sprintf', $args);
+    }
+}
+
+if (!function_exists('strEnds')) {
+    /**
+     * Returns whether or not a string ends with another string.
+     *
+     * This function is not case-sensitive.
+     *
+     * @param string $haystack The string to test.
+     * @param string $needle The substring to test against.
+     * @return bool Whether or not `$string` ends with `$with`.
+     * @category String Functions
+     */
+    function strEnds($haystack, $needle) {
```

Comment:
Sorry about this, but can we make this a private method of Schema instead? I want to reduce dependencies as much as possible.